src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/ReferenceCleanupService.ReferenceHandlers.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.VS.References.Roslyn;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.References
+{
+    internal partial class ReferenceCleanupService
+    {
+        private abstract class ReferenceHandler
```

Comment:
It's not uncommon to name abstract base classes with a `Base` suffix.

Reply:
We copied/were forced to use **Abstract**XXX pattern from Roslyn.

Reply:
done. Changed to match pattern from Roslyn